AI/VCAI/Engine/PriorityEvaluator.cpp

```diff
@@ -0,0 +1,354 @@
+		value->addTerm(new fl::Triangle("BITHIGH", 0.650, 0.800, 0.800, 0.250));
+		value->addTerm(new fl::Triangle("HIGH", 0.750, 0.900, 0.900, 0.500));
+		value->addTerm(new fl::Ramp("HIGHEST", 0.850, 1.000));
+		value->setRange(0.0, 1.0);
+
+		//we may want to use secondary hero(es) rather than main hero
+
+		//do not cancel important goals
+		//addRule("if lockedMissionImportance is HIGH then Value is very LOW");
+		//addRule("if lockedMissionImportance is MEDIUM then Value is somewhat LOW");
+		//addRule("if lockedMissionImportance is LOW then Value is HIGH");
+
+		//pick nearby objects if it's easy, avoid long walks
+		/*addRule("if turnDistance is SMALL then Value is somewhat HIGH");
+		addRule("if turnDistance is MEDIUM then Value is MEDIUM");
+		addRule("if turnDistance is LONG then Value is LOW");*/
+
+		//some goals are more rewarding by definition f.e. capturing town is more important than collecting resource - experimental
+		addRule("if turnDistance is MEDIUM then Value is MEDIUM");
+		addRule("if turnDistance is SMALL then Value is BITHIGH");
+		addRule("if turnDistance is LONG then Value is BITLOW");
+		addRule("if turnDistance is very LONG then Value is LOW");
+		addRule("if goldReward is LOW then Value is MEDIUM");
+		addRule("if goldReward is MEDIUM and armyLoss is LOW then Value is BITHIGH");
+		addRule("if goldReward is HIGH and armyLoss is LOW then Value is HIGH");
+		addRule("if armyReward is LOW then Value is MEDIUM");
+		addRule("if armyReward is MEDIUM and armyLoss is LOW then Value is BITHIGH");
+		addRule("if armyReward is HIGH and armyLoss is LOW then Value is HIGHEST with 0.5");
+		addRule("if armyReward is HIGH and goldReward is HIGH and armyLoss is LOW then Value is HIGHEST");
+		addRule("if armyReward is HIGH and goldReward is MEDIUM and armyLoss is LOW then Value is HIGHEST with 0.8");
+		addRule("if armyReward is MEDIUM and goldReward is HIGH and armyLoss is LOW then Value is HIGHEST with 0.5");
+		addRule("if armyReward is MEDIUM and goldReward is MEDIUM and armyLoss is LOW then Value is HIGH");
+		addRule("if armyReward is HIGH and turnDistance is SMALL and armyLoss is LOW then Value is HIGHEST");
+		addRule("if goldReward is HIGH and turnDistance is SMALL and armyLoss is LOW then Value is HIGHEST");
+		addRule("if armyReward is MEDIUM and turnDistance is SMALL and armyLoss is LOW then Value is HIGH");
+		addRule("if goldReward is MEDIUM and turnDistance is SMALL and armyLoss is LOW then Value is BITHIGH");
+		addRule("if goldReward is LOW and armyReward is LOW and turnDistance is not SMALL then Value is LOWEST");
+		addRule("if armyLoss is HIGH then Value is LOWEST");
+		addRule("if armyLoss is LOW then Value is MEDIUM");
+		addRule("if armyReward is LOW and turnDistance is LONG then Value is LOWEST");
+		addRule("if danger is NONE and armyStrength is HIGH and armyReward is LOW then Value is LOW");
+		addRule("if danger is NONE and armyStrength is HIGH and armyReward is MEDIUM then Value is BITLOW");
```

Comment:
Other fuzzy logic terms for army Strength are unused in rules. That makes output imbalanced as engine does not react at all when armyStrength is at range 0.2 - 0.5. If you do not want to make counter-rules that impact output for LOW / MEDIUM then consider removing rest of terms, then giving HIGH whole range (0-1) and change term shape.

Reply:
Looks like it will evaluate in heroType MAIN/SECONDARY/SCOUT. Secondary is under the question. Maybe it will by subtype of main. Scouts will have different rules priority evaluating.